### PR TITLE
Feat/upgrade to 0.1.3

### PR DESCRIPTION
**Pull request by efiege**
# Pull Request

_Briefly describe WHAT your PR changes, which features it adds/modifies._

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

-   Test A
-   Test B
-   ...

**Test Configuration**:

-   Firmware version:
-   Hardware:
-   Toolchain:
-   SDK:

## Linked Issue(s)

_Use keywords to automate: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword_

-   fixes # (issue)
-   closes # (issue)
-   ...

## PR is blocked by

-   [ ] blocked by # (issue)

# Checklist

-   [ ] I have **formatted the title** correctly and precisely
-   [ ] My code follows the **style guidelines** of this project
-   [ ] I have performed a **self-review** of my own code
-   [ ] I have **commented** my code, particularly in hard-to-understand areas and public classes/methods
-   [ ] I have made corresponding changes to the **documentation**
-   [ ] My changes generate **no new warnings** (performed checkstyle check locally)
-   [ ] I have added **tests that prove my fix** is effective or that my feature works
-   [ ] New and existing unit **tests pass locally** with my changes
-   [ ] Any dependent **changes have been merged** and published in downstream modules
-   [ ] I have added/updated **copyright headers**

### CODE REVIEW

File: docker-compose-dev.yaml [hunk 1]
```diff
@@ -39,14 +44,18 @@ services:
...
 
       MY_EDC_JDBC_URL: jdbc:postgresql://postgresql:5432/edc
       MY_EDC_JDBC_USER: edc
       MY_EDC_JDBC_PASSWORD: edc
       EDC_WEB_REST_CORS_ENABLED: 'true'
       EDC_WEB_REST_CORS_HEADERS: 'origin,content-type,accept,authorization,x-api-key'
       EDC_WEB_REST_CORS_ORIGINS: '*'
+
+      EDC_PARTICIPANT_ID: provider
+      EDC_JSONLD_HTTPS_ENABLED: true
```
Comment (richardtreier) [inline docker-compose-dev.yaml:57]: should go to .env

File: docker-compose-dev.yaml [hunk 2]
```diff
@@ -39,14 +44,18 @@ services:
...
+      MY_EDC_PROTOCOL_BASE_URL: "http://edc:11003"
 
       MY_EDC_JDBC_URL: jdbc:postgresql://postgresql:5432/edc
       MY_EDC_JDBC_USER: edc
       MY_EDC_JDBC_PASSWORD: edc
       EDC_WEB_REST_CORS_ENABLED: 'true'
       EDC_WEB_REST_CORS_HEADERS: 'origin,content-type,accept,authorization,x-api-key'
       EDC_WEB_REST_CORS_ORIGINS: '*'
+
+      EDC_PARTICIPANT_ID: provider
```
Comment (richardtreier) [inline docker-compose-dev.yaml:56]: should be moved to .env and sourced from MY_EDC_ID_KEBAP_CASE or whatever it's called

File: docker-compose-dev.yaml [hunk 3]
```diff
@@ -39,14 +44,18 @@ services:
...
+      EDC_JSONLD_HTTPS_ENABLED: true
+      EDC_DSP_CALLBACK_ADDRESS: http://edc:11003/api/v1/protocol
```
Comment (richardtreier) [inline docker-compose-dev.yaml:58]: should be generated in .env from MY_EDC_PROTOCOL_BASE_URL

File: docker-compose-dev.yaml [hunk 4]
```diff
@@ -7,11 +7,16 @@ services:
...
-      - EDC_UI_CATALOG_URLS=http://edc2:11003/api/v1/ids/data
+      - EDC_UI_CATALOG_URLS=http://edc2:11003/api/v1/protocol
   edc:
-    image: ${DEV_EDC_IMAGE}
+#    image: ${DEV_EDC_IMAGE}
+    build:
+      dockerfile: connector/Dockerfile
+      args:
+        JVM_ARGS: -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005
+        BUILD_ARGS: -Pfs-vault -Pdmgmt-api-key -Ppostgres-flyway -Poauth
```
Comment (richardtreier) [inline docker-compose-dev.yaml:19]: remove from PR

File: e2e-test/src/test/java/de/sovity/edc/extension/e2e/PostgresFlywayExtensionTest.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+import static org.eclipse.edc.connector.transfer.spi.types.TransferProcessStates.COMPLETED;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.CONTEXT;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.TYPE;
+import static org.eclipse.edc.jsonld.spi.PropertyAndTypeNames.ODRL_POLICY_ATTRIBUTE;
+
+@Disabled("Just to be executed with postgres-flyway property set")
+public class PostgresFlywayExtensionTest {
+
+    private static final String PROVIDER_TARGET_URL = "https://google.de";
+    private static final String CONSUMER_BACKEND_URL = "https://webhook.site/e542f69e-ff0a-4771-af18-0900a399137a";
```
Comment (richardtreier) [inline e2e-test/src/test/java/de/sovity/edc/extension/e2e/PostgresFlywayExtensionTest.java:33]: is this ok?

File: e2e-test/src/test/java/de/sovity/edc/extension/e2e/PostgresFlywayExtensionTest.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+import static de.sovity.edc.extension.e2e.connector.config.EdcApiType.PROTOCOL;
+import static jakarta.json.Json.createObjectBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.eclipse.edc.connector.transfer.spi.types.TransferProcessStates.COMPLETED;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.CONTEXT;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.TYPE;
+import static org.eclipse.edc.jsonld.spi.PropertyAndTypeNames.ODRL_POLICY_ATTRIBUTE;
+
+@Disabled("Just to be executed with postgres-flyway property set")
```
Comment (richardtreier) [inline e2e-test/src/test/java/de/sovity/edc/extension/e2e/PostgresFlywayExtensionTest.java:29]: ???

File: e2e-test/src/test/java/de/sovity/edc/extension/e2e/connector/factory/ConnectorFactory.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package de.sovity.edc.extension.e2e.connector.factory;
+
+import de.sovity.edc.extension.e2e.connector.Connector;
+
+public interface ConnectorFactory {
```
Comment (richardtreier) [inline e2e-test/src/test/java/de/sovity/edc/extension/e2e/connector/factory/ConnectorFactory.java:5]: why is this an interface

File: e2e-test/src/test/java/de/sovity/edc/extension/e2e/connector/factory/TestConnectorFactory.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    @Override
+    public Connector createConnector() {
+        var apiConfigMap = createApiConfigMap();
+        var dspCallbackAddress = apiConfigMap.get(PROTOCOL).getUri();
+        var datasourceConfigs = getDatasourceConfigs(testDatabase);
+        var connector = TestConnector.builder()
+                .participantId(participantId)
+                .apiConfigMap(apiConfigMap)
+                .datasourceConfigs(datasourceConfigs)
+                .simpleConfig(new SimpleConfig("edc.participant.id", participantId))
```
Comment (richardtreier) [inline e2e-test/src/test/java/de/sovity/edc/extension/e2e/connector/factory/TestConnectorFactory.java:50]: withConfig(String, String)?
~~withConfig(Map<String, String>)~~

File: extensions/postgres-flyway/src/main/resources/migration/consumer/V1__m8-consumer-sample-data.sql [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+INSERT INTO public.edc_data_request (datarequest_id, process_id, connector_address, protocol, connector_id, asset_id, contract_id, data_destination, managed_resources, properties, transfer_type, transfer_process_id) VALUES ('5e4cb5c8-fc8e-4231-ba86-bc5dbf7b5015', '683098a6-0ed0-4b1a-bf19-f1b0dc67f682', 'http://edc:11003/api/v1/ids/data', 'ids-multipart', 'consumer', 'urn:artifact:test:1.0', 'test-contract-definition:2160b70f-d65e-4861-af88-f29b16f3fa7b', '{"properties":{"baseUrl":"https://webhook.site/e542f69e-ff0a-4771-af18-0900a399137a","method":"POST","type":"HttpData"}}', false, '{}', '{"contentType":"application/octet-stream","isFinite":true}', '683098a6-0ed0-4b1a-bf19-f1b0dc67f682');
```
Comment (richardtreier) [inline extensions/postgres-flyway/src/main/resources/migration/consumer/V1__m8-consumer-sample-data.sql:84]: data should contain both a providing and a consuming data offer (both with transfers)

File: extensions/postgres-flyway/src/main/resources/migration/default/V2__MS8-to-0.1.0.sql [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+-- Update Asset IDs
+ALTER TABLE edc_asset_dataaddress DROP CONSTRAINT edc_asset_dataaddress_asset_id_fk_fkey;
+ALTER TABLE edc_asset_property DROP CONSTRAINT edc_asset_property_asset_id_fk_fkey;
+UPDATE edc_asset SET asset_id = REPLACE(REPLACE(asset_id, 'urn:artifact:', ''), ':', '-');
+UPDATE edc_asset_dataaddress SET asset_id_fk = REPLACE(REPLACE(asset_id_fk, 'urn:artifact:', ''), ':', '-');
+UPDATE edc_asset_property SET asset_id_fk = REPLACE(REPLACE(asset_id_fk, 'urn:artifact:', ''), ':', '-');
+ALTER TABLE edc_asset_dataaddress ADD CONSTRAINT edc_asset_dataaddress_asset_id_fk_fkey FOREIGN KEY (asset_id_fk) REFERENCES edc_asset (asset_id) ON DELETE CASCADE;
+ALTER TABLE edc_asset_property ADD CONSTRAINT edc_asset_property_asset_id_fk_fkey FOREIGN KEY (asset_id_fk) REFERENCES edc_asset (asset_id) ON DELETE CASCADE;
+
+UPDATE edc_asset_property SET property_name = 'https://w3id.org/edc/v0.0.1/ns/id', property_value=asset_id_fk WHERE property_name = 'asset:prop:id';
```
Comment (richardtreier) [inline extensions/postgres-flyway/src/main/resources/migration/default/V2__MS8-to-0.1.0.sql:27]: starting 0.1.x all asset properties need to be URLs:

- MDS properties are already URLs.
- All asset:prop: fields need to be migrated.
- Fields like "description", "keyword", "langauge" might have EDC equivalents
- Custom fields might not, then we need to prepend a custom namespace.

File: gradle.properties [hunk 11]
```diff
@@ -2,7 +2,7 @@ sovityEdcExtensionsVersion=0.0.1-SNAPSHOT
 sovityEdcExtensionGroup=de.sovity.edc.ext
 sovityEdcGroup=de.sovity.edc
 edcGroup=org.eclipse.edc
-edcVersion=0.0.1-20230220.patch1
+edcVersion=0.1.3
```
Comment (richardtreier) [inline gradle.properties:5]: da geht noch mehr

**Comment by tmberthold** [comment]
Housekeeping - this PR can be closed right?

**Pull request closed without merging**